src/wasm.cc

```diff
@@ -446,6 +446,32 @@ void WasmBase::finishShutdown() {
   }
 }
 
+bool canary(std::shared_ptr<WasmHandleBase> &wasm_handle, const std::shared_ptr<PluginBase> &plugin,
```

Comment:
Does it make sense to make canary(...) a member function of WasmHandle?

Reply:
That makes sense. I have changed it.